### PR TITLE
Simple indexed attribute access in station files

### PR DESCRIPTION
**Pull request by hp-peti**
Some drivers (see zurich-instruments) only expose parameters through an interface which needs indexing in addition to attribute access. This patch enables access to those parameters in the station yaml file.

### CODE REVIEW

File: src/qcodes/station.py
Comment (jenshnielsen) [inline src/qcodes/station.py:600]: Could we extend the docstring above with an example of what this adds

I guess something like ?

dac.channels[0]

**Comment by jenshnielsen** [comment]
Would be nice with an example of this in the station notebook here https://github.com/microsoft/Qcodes/blob/main/docs/examples/Station.ipynb  but I think that is optional

**Review by jenshnielsen — APPROVED** [review]

File: src/qcodes/station.py
Comment (hp-peti) [reply src/qcodes/station.py:600]: on it

**Comment by jenshnielsen** [comment]
@hp-peti BTW unless there are conflicts you do not need to worry about merging main into the feature branch. Github actions will take care of making sure that it is tested against main before merging

**Comment by hp-peti** [comment]
need workflow approval

> Would be nice with an example of this in the station notebook here https://github.com/microsoft/Qcodes/blob/main/docs/examples/Station.ipynb but I think that is optional.
I don't have the courage to modify that notebook; the example station.yaml is an image file i don't want to mess with.
> 
> Other than that the only remaining issue is IMHO the precommit hooks
auto organized utils/__init__.py to fix issue

**Comment by hp-peti** [comment]
need another workflow approval. 

not sure what's happening on the merge queue. 
@jenshnielsen let me know there is anything i need to do
thanks in adv

**Comment by hp-peti** [comment]
pinging again, i need workflow approval and please put back in merge queue. thanks in advance

**Comment by hp-peti** [comment]
thanks everyone

**Pull request merged**
